Erick Assertion Handler

LinnierGames opened the pull request:
- [X] Compiles on desktop
- [x] Complies on Teensy
- [x] Complete Codes SpreadSheet

closes #83 

Substitute `assert(e)` with `assertion( condition, integer code )`. Definitions of code values is saved in [AssertionHandler-Dictionary.csv](https://docs.google.com/spreadsheets/d/1zIhQ-8UmOYflNnDYCzlLkUQ36gXbFe5Jb5eG3tSFJYs/edit#gid=0)
#### Changes

( + ) AssertionHandler.h
- using a macro to output to the console using Logger( .. )

LinnierGames committed:
init Xcode

LinnierGames committed:
import files into project

LinnierGames committed:
noone

LinnierGames committed:
assertion prototype

LinnierGames committed:
Merge branch 'dev' into erick_assertion-handler

LinnierGames committed:
fix : merges

updated main.cpp

LinnierGames committed:
Read and write to Memory

blhough commented on micromouse-xcode/micromouse-xcode.xcodeproj/project.pbxproj:
```diff
@@ -0,0 +1,346 @@
+	objects = {
+
+/* Begin PBXBuildFile section */
+		E1B489D01CC87D9300643A24 /* main.cpp in Sources */ = {isa = PBXBuildFile; fileRef = E1B489CF1CC87D9300643A24 /* main.cpp */; };
```
wow xcode projects are ugly

Commit:
remove xcode project files

LinnierGames committed:
fix : typo

LinnierGames committed:
Merge branch 'erick_assertion-memory' into erick_assertion-handler

LinnierGames committed:
Merge branch 'dev' into erick_assertion-handler

# Conflicts:
#	workspace.xcworkspace/contents.xcworkspacedata

LinnierGames committed:
Removed micromouse-xcode Folder

LinnierGames committed:
Extracting Codes

Codes are located at this
[spreadsheet](https://docs.google.com/spreadsheets/d/1zIhQ-8UmOYflNnDYCz
lLkUQ36gXbFe5Jb5eG3tSFJYs/edit?usp=sharing)

LinnierGames committed:
Merge remote-tracking branch 'origin/erick_assertion-handler' into erick_assertion-handler

# Conflicts:
#	workspace.xcworkspace/contents.xcworkspacedata

blhough commented on micromouse/AssertionHandler.h:
```diff
@@ -0,0 +1,45 @@
+    Micromouse::Memory::write( Micromouse::ERROR_MEMORY, intNextIndex);
+    
+}
+
```
can you add a function here that dumps all the errors in memory to the serial port using log()

LinnierGames replied:
Hmm, you mean read all the codes and output them to the serial port?

blhough replied:
yes

LinnierGames replied:
I can't tell if i did it right :P

blhough replied:
Not quite, look at logger.h for proper use

LinnierGames replied:
how about now???
plus i can't tell why i can't call to this `inline` method anywhere in the project :P

blhough commented:
Looks good, just need a way to view the errors

LinnierGames committed:
log() implementation for AssertionHandler

LinnierGames committed:
fix : use of log( .. )

blhough commented on micromouse/AssertionHandler.h:
```diff
@@ -29,7 +30,7 @@ inline static void logError( int code ) {
 inline static void printErrorCodes() {
```
Why does it need to be inline?

LinnierGames replied:
hmmm, beats the hell out of me, i thought i had to, does it need to be static? i would say no huh

blhough replied:
no, it can just be a regular global function

blhough commented:
Needs to be tested on the mouse
